File: Flutter/lib/core/network/api_routes.dart
class ApiRoutes {
  ApiRoutes._();

  static const String apiPrefix = '/api/v1';

  static const String root = apiPrefix;
  static const String ping = '$apiPrefix/ping';
  static const String appVersion = '$apiPrefix/app-version';

  static const String authCallbackPath = '/auth/callback';
  static const String inviteCallbackPath = '/auth/invite-callback';
  static const String authDeviceToken = '$apiPrefix/auth/device-token';
  static const String authInvite = '$apiPrefix/auth/invite';
  static const String authLoginRiskCheck = '$apiPrefix/auth/login-risk-check';
  static const String authLoginAttempt = '$apiPrefix/auth/login-attempt';
  static const String authSessionSync = '$apiPrefix/auth/session-sync';
  static const String authRefresh = '$apiPrefix/auth/refresh';

  static const String mediaUploadUrl = '$apiPrefix/media/upload-url';
  static const String mediaComplete = '$apiPrefix/media/complete';
  static const String mediaFile = '$apiPrefix/media/file';

  static const String aiQuestion = '$apiPrefix/ai/question';

  static const String notifyPartner = '$apiPrefix/notify/partner';

  static const String userWipe = '$apiPrefix/users/wipe';
}

File: Backend/config/apiRoutes.js
const API_PREFIX = "/api";
const API_V1_PREFIX = `${API_PREFIX}/v1`;

const CALLBACK_ROUTES = {
  auth: "/auth/callback",
  invite: "/auth/invite-callback",
};

const API_V1_PATHS = {
  root: API_V1_PREFIX,
  ping: `${API_V1_PREFIX}/ping`,
  authBase: `${API_V1_PREFIX}/auth`,
  mediaBase: `${API_V1_PREFIX}/media`,
  aiBase: `${API_V1_PREFIX}/ai`,
  notifyBase: `${API_V1_PREFIX}/notify`,
  usersBase: `${API_V1_PREFIX}/users`,
  appVersion: `${API_V1_PREFIX}/app-version`,
  authRefresh: `${API_V1_PREFIX}/auth/refresh`,
  authLoginAttempt: `${API_V1_PREFIX}/auth/login-attempt`,
  authLoginRiskCheck: `${API_V1_PREFIX}/auth/login-risk-check`,
  authSessionSync: `${API_V1_PREFIX}/auth/session-sync`,
  authInvite: `${API_V1_PREFIX}/auth/invite`,
  authDeviceToken: `${API_V1_PREFIX}/auth/device-token`,
  aiQuestion: `${API_V1_PREFIX}/ai/question`,
  mediaFile: `${API_V1_PREFIX}/media/file`,
  userWipe: `${API_V1_PREFIX}/users/wipe`,
};

module.exports = {
  API_PREFIX,
  API_V1_PREFIX,
  CALLBACK_ROUTES,
  API_V1_PATHS,
};
